packages/pigeon/lib/dart_generator.dart

```diff
@@ -500,48 +512,48 @@ void generateDart(DartOptions opt, Root root, StringSink sink) {
       indent.write('Object encode() ');
       indent.scoped('{', '}', () {
         indent.writeln(
-          'final Map<Object?, Object?> pigeonMap = <Object?, Object?>{};',
+          'final List<Object?> pigeonList = <Object?>[];',
         );
         for (final NamedType field in klass.fields) {
```

Comment:
I think we should extract a cross-generator helper (eventually in the skeleton, but it's fine to just put it in the existing helpers file for now) to return an object's fields in serialization/deserialization order. Even though it's just `klass.fields`, it's critical that it be the same across all generators so I think it would be good from a safety/maintainability standpoint to have everywhere we are relying on that in every generator calling something like `Iterator<NamedType> getFieldsInSerializationOrder(Class klass)` so it's very obvious where those consistency points are.

Reply:
I'll add that. I wasn't sure if it was needed, since in my testing there was a 100% correct order rate for this. Can't argue against safety though!

Reply:
Mostly my reasoning here is:
- it gives us a clear point to change if we want to change it in the future, and
- it will make it very clear that someone changing the code shouldn't do something like say "hey, the generated output would be nicer looking if this were in alphabetical order, let's sort it!"

Arguably the latter isn't actually worth worrying about since integration tests would all explode, but I'm a fan of expressing sementic intent with helper functions.

Reply:
I've added a version of this function that basically doesn't do anything atm. Let me know if there's more I should do with it. Having a hard time thinking of a way to make it better.